app/src/main/java/org/mozilla/fenix/settings/wallpaper/Extensions.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.wallpapers
+
+/**
+ * The extension function to group wallpapers according to their name.
+ **/
+
+fun List<Wallpaper>.groupByCollection(): Map<Wallpaper.Collection, List<Wallpaper>> = groupBy {
```

Comment:
Might be nice to add a unit test for this

Reply:
Just as a side note, wouldn't it make sense to have a Collection object having a list of wallpapers inside? We don't really need that `Map<Wallpaper.Collection, List<Wallpaper>>` map, do we?